### ISSUE TITLE
Update the example for SLP Minting baton contract

### ISSUE THREAD

**Issue by p0o** [Bitcoin-com/cashscript#77]
[This contract](https://github.com/simpleledgerinc/slp-mint-contracts/blob/master/slp-mint-vault-v0.cash) is for 11 months ago and I'm not even sure if it works with the latest version of the CashScript?

However with the latest changes in Cash Script I think it would be great to maintain a version of that contract in the examples folder with the new way of using preimage and every other changes that is required to be done.

What do you think @rkalis? If you are open I wouldn't mind giving it a try myself but need some help in review

**Comment by rkalis** [comment]
I agree with that, and would be happy to help out with review. I think ideally this project would be picked up and expanded separately from this repo, but I don't think that @jcramer has the resources to maintain that, so I think having it as an updated example in the CashScript repo is a good solution.

**Comment by p0o** [comment]
Cool, I will be preparing something soon. That contract is still a bit ambiguous for me, the question is we still need a central pk to issue the token using the contract right?

**Comment by rkalis** [comment]
Yes exactly, this first version of the contract *only* makes sure that the minting baton cannot be accidentally burned, but otherwise is still custodial.

**Comment by jcramer** [comment]
I am working on an SLP Vault contract in my fork of cashscript. For the past few days have been just trying to get it baked into EC SLP wallet with a nice UX.  

Perhaps the SLP Vault will also handle the hardening of a mint baton also.  maybe it's it will be it's own contact, not sure yet.

I want to make the contract in the form of p2sh = f(phh) similar to the current slp vault so that ones normal hd can easily detect ownership.

**Comment by p0o** [comment]
@jcramer That's great, any plans to provide a cash script v0.5.0 compatible version of it here or in your repo for others to use?

**Comment by jcramer** [comment]
I can try. Is 0.5.0 compatibility important for some reason?

**Comment by p0o** [comment]
I proposed it only for readability since the API is more abstract and there is no SLP example available here. I made one to work with SLP with 0.5.0 API [here](https://github.com/p0o/yield-farming-bch-smart-contract) but this one is minting with a different mechanism.
